rust/flatc/src/lib.rs

```diff
@@ -0,0 +1,281 @@
+        self
+    }
+
+    /// Add include directories to the flatc compilation.
+    ///
+    /// The order in which directories are added will be preserved.
+    ///
+    /// ## Shell Command
+    ///
+    /// ```text
+    /// $ flatc -r -I dir1 -I dir2 foo.fbs
+    /// ```
+    ///
+    /// ## Rust equivalent
+    ///
+    /// ```no_run
+    /// extern crate flatc;
+    ///
+    /// fn main() {
+    ///     flatc::Build::new()
+    ///         .schema("foo.fbs")
+    ///         .include("dir1")
+    ///         .include("dir2")
+    ///         .compile();
+    /// }
+    /// ```
+    pub fn include<P: AsRef<Path>>(&mut self, dir: P) -> &mut Self {
+        self.include_paths.push(dir.as_ref().to_path_buf());
+        self
+    }
+
+    /// Run the FlatBuffer schema compiler, generating one file for each of the
+    /// schemas added to the `Build` with the
+    /// [`schema`](struct.Build.html#method.schema) method.
+    ///
+    /// Output filenames are postfixed with "*_generated*", and the extension is
+    /// set to "*.rs*". These files are placed in `OUT_DIR`. They can be
+    /// included in your crate using compile time macros.
+    ///
+    /// # Example
+    ///
+    /// For a compilation of `foo.fbs`, you would include it in your main crate
+    /// with:
+    ///
+    /// ```ignore
+    /// mod foo_generated {
+    ///     include!(concat!(env!("OUT_DIR"), "/foo_generated.rs"));
+    /// }
+    /// ```
+    ///
+    /// # Panics
+    ///
+    /// This function will panic if it encounters any error. This will be
+    /// emitted as a compilation error when compiling your crate.
+    pub fn compile(&self) {
+        if let Err(e) = self.try_compile() {
+            panic!("\n\n{}\n\n", e);
+        }
+    }
+}
+
+impl Build {
+    fn try_compile(&self) -> Result<(), Error> {
+        let output_path = self.output_path.as_ref().map_or_else(
+            || env::var("OUT_DIR").unwrap(),
+            |path| format!("{}", path.display()),
+        );
+
+        let include_paths = self
+            .include_paths
+            .iter()
+            .map(|path| format!("-I {}", path.display()));
+
+        let fbs_files = self.schema.iter();
+
+        Command::new(FLATC_EXECUTABLE)
+            .current_dir(env::var("CARGO_MANIFEST_DIR").unwrap())
+            .arg("-r")
+            .args(&["-o", &output_path])
+            .args(include_paths)
+            .args(fbs_files)
+            .output()
+            .map_err(Error::from)
+            .and_then(|output| {
+                if output.status.success() {
+                    Ok(())
+                } else {
+                    Err(Error::new(
+                        ErrorKind::FlatC,
+                        String::from_utf8(output.stdout).unwrap().trim(),
+                    ))
+                }
+            })
+    }
+}
+
+impl Build {
+    #[doc(hidden)]
+    pub fn output<P: AsRef<Path>>(&mut self, dir: P) -> &mut Self {
```

Comment:
Not happy with having this as part of the public interface and simply hidden from the docs (as it would permit misuse of `Builder`), but we need to do this to test the crate from `/tests/rust_usage_test`.